cli/src/infragpt/llm/providers/anthropic_provider.py

```
@@ -0,0 +1,211 @@
+                            tool_call = ToolCall(
+                                id=tool_block["id"],
+                                name=tool_block["name"],
+                                arguments=arguments
+                            )
+                            yield StreamChunk(tool_calls=[tool_call])
+                                
+                        except json.JSONDecodeError as e:
+                            print(f"Warning: Failed to parse JSON for {tool_block['name']}: {json_str} - Error: {e}")
+                
+                elif event.type == "message_delta":
+                    if hasattr(event.delta, 'stop_reason'):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)